types/modules/core/firebase.d.ts

```diff
@@ -0,0 +1,59 @@
+import App from './firebase-app';
+import AdMob from '../admob';
+import Auth from '../auth';
+import Analytics from '../analytics';
+import Config from '../config';
+import Crash from '../crash';
+import Database from '../database';
+import Firestore from '../firestore';
+import Links from '../links';
+import Messaging from '../messaging';
+import Perf from '../perf';
+import Storage from '../storage';
+import Utils from '../utils';
+import { FirebaseOptions } from '../../types/index';
+export declare class Firebase {
+    admob: AdMob;
```

Comment:
I don't think these declarations are correct - it should export both the module constructor and the set of statics.

Reply:
Heh, yeah, I found that out pretty quickly while testing (https://github.com/meyer/react-native-firebase/commit/c5e157e9f06275c0f7ac6892020ac003dcb2e49d). Still trying to figure out the best way to mix statics object with the function in an automated/inferred way. Worst case scenario would be specifying them manually, but that would lead to eventual mismatches.